app/src/main/java/com/cp/tasky/auth/login/presentation/LoginViewModel.kt

```diff
@@ -29,8 +28,8 @@ class LoginViewModel @Inject constructor(
     var screenState by mutableStateOf(LoginScreenState())
         private set
 
-    private val _networkState = MutableStateFlow<Result<LoginResponse, Error>>(Result.Idle)
-    val networkState = _networkState.asStateFlow()
+    private val _networkResponse = Channel<Result<Unit, Error>>(Channel.BUFFERED)
```

Comment:
Instead of making this channel for the whole network response, keep it light and just send the information the UI really needs to know (here just that login was successful). Typically, you work with a sealed interface for these types of events as well (to avoid confusion I call UI -> ViewModel actions and ViewModel -> UI events):
```
sealed interface LoginEvent {
    data object LoginSuccess: LoginEvent
    data class Error(val error: UiText): LoginEvent
}
```
Not sure if you know my UiText class, but it's pretty much just a wrapper around strings and string resources to already wrap them in the VM:
```
sealed class UiText {
    data class DynamicString(val value: String): UiText()
    class StringResource(@StringRes val id: Int, vararg val args: Any): UiText()

    companion object {
        fun unknownError(): UiText {
            return UiText.StringResource(R.string.error_unknown)
        }
    }

    fun asString(context: Context): String {
        return when(this) {
            is DynamicString -> value
            is StringResource -> context.getString(id, *args)
        }
    }
    
    @Composable
    fun asString(): String {
        return when(this) {
            is DynamicString -> value
            is StringResource -> LocalContext.current.getString(id, *args)
        }
    }
}
```